### PR TITLE
feat(PLAT-3087): zip reader-writer

### PR DESCRIPTION
**Pull request by mustyantsev**
Implement reading zip (tdf) file from the end

### CODE REVIEW

File: sdk/src/test/java/io/opentdf/platform/sdk/ZipWriterTest.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+
+public class ZipWriterTest {
```
Comment (sujankota) [inline sdk/src/test/java/io/opentdf/platform/sdk/ZipWriterTest.java:12]: Can we add tests for adding more files to zip and also some tests with large data?

Comment (sujankota) [reply sdk/src/test/java/io/opentdf/platform/sdk/ZipWriterTest.java:12]: Same thing for ZipReader - Reading large zip files.

**Review by sujankota — APPROVED** [review]

File: sdk/pom.xml [hunk 2]
```diff
@@ -57,5 +57,11 @@
             <version>5.0.0-alpha.14</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
```
Comment (mkleene) [inline sdk/pom.xml:60]: we are using [junit 5](https://github.com/opentdf/java-sdk/blob/cab573ad79d21f27e06ef244819d1001cd469a10/sdk/pom.xml#L20-L23) so I don't think that we need this import.

File: sdk/src/main/java/io/opentdf/platform/sdk/ZipReader.java [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+                break;
+            }
+            pointer--;
+        }
+
+        if (pointer < 0) {
+            throw new Exception("Invalid tdf file");
+        }
+
+        // Read the EOCDR
```
Comment (mkleene) [inline sdk/src/main/java/io/opentdf/platform/sdk/ZipReader.java:42]: this probably doesn't matter but are these values signed?

File: sdk/src/main/java/io/opentdf/platform/sdk/ZipReader.java [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+        short centralDirectoryDiskNumber = buffer.getShort();
+        short numEntriesThisDisk = buffer.getShort();
+        numEntries = buffer.getShort();
+        int centralDirectorySize = buffer.getInt();
+        offsetToStartOfCentralDirectory = buffer.getInt();
+        short commentLength = buffer.getShort();
+
+        // buffer's position at the start of the Central Directory
+        boolean isZip64 = false;
+        if (offsetToStartOfCentralDirectory != ZIP64_MAGICVAL) {
```
Comment (mkleene) [inline sdk/src/main/java/io/opentdf/platform/sdk/ZipReader.java:53]: maybe invert this condition?

File: sdk/src/test/java/io/opentdf/platform/sdk/ZipReaderTest.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+public class ZipReaderTest {
+    private ZipReader zipReader;
+    private ByteBuffer buffer;
+    private RandomAccessFile raf;
+    private FileChannel fileChannel;
+
+
+    @Test
+    public void testZipReader() throws Exception {
+        RandomAccessFile raf = new RandomAccessFile("src/test/resources/sample.txt.tdf", "r");
```
Comment (mkleene) [inline sdk/src/test/java/io/opentdf/platform/sdk/ZipReaderTest.java:24]: maybe read this as a resource so that it's insensitive to the working directory?

**Review by mkleene — APPROVED** [review]

**Pull request merged**